File: sources/Android/android-communications/src/main/java/polar/com/sdk/api/PolarBleApiDefaultImpl.java
// Copyright © 2019 Polar Electro Oy. All rights reserved.
package polar.com.sdk.api;

import android.content.Context;

import androidx.annotation.NonNull;

import polar.com.sdk.impl.BDBleApiImpl;

/**
 * Simply returns a new default implementation of the API.
 */
public class PolarBleApiDefaultImpl {
    /**
     * Default implementation constructor for the API.
     *
     * @param context  where API implementation callbacks are run
     * @param features @see polar.com.sdk.api.PolarBleApi feature flags
     * @return default Polar API implementation
     */
    @NonNull
    public static PolarBleApi defaultImplementation(final @NonNull Context context, int features) {
        return new BDBleApiImpl(context, features);
    }

    /**
     * @return SDK version number in format major.minor.patch
     */
    @NonNull
    public static String versionInfo() {
        return "3.1.1";
    }
}

File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/PolarBleApiDefaultImpl.swift
/// Copyright © 2019 Polar Electro Oy. All rights reserved.

import Foundation

/// Class to provide the default implementation of the Polar Ble API
public class PolarBleApiDefaultImpl {
    
    /// New instance of Polar Ble API implementation
    ///
    /// - Parameter queue: context of where the API is used
    /// - Parameter features: set of SDK features to enable. Pass an empty set to enable all SDK features by default.
    /// - Parameter restoreIdentifier: optional Core Bluetooth state restoration identifier for background relaunch support
    /// - Returns: api instance
    public static func polarImplementation(_ queue: DispatchQueue, features: Set<PolarBleSdkFeature>, restoreIdentifier: String? = nil) -> PolarBleApi {
        return PolarBleApiImpl(queue, features: features, restoreIdentifier: restoreIdentifier)
    }
    
    /// Return current version
    ///
    /// - Returns: version in format major.minor.patch
    public static func versionInfo() -> String {
        return "8.0.0"
    }
}
